app/javascript/packs/dashboard.ts

```
@@ -1,7 +1,41 @@
+// Create router instance
+const router = createDashboardRouter();
+
+// Initialize SPA when DOM is ready
+document.addEventListener("DOMContentLoaded", () => {
+  const rootElement = document.getElementById("dashboard-root");
+  
+  if (rootElement) {
+    const root = ReactDOM.createRoot(rootElement);
+    
+    root.render(
+      React.createElement(DashboardProvider, { children: 
```

Biome (2.1.2)

[error] Avoid passing children using a prop

The canonical way to pass children in React is to use additional arguments to React.createElement

(lint/correctness/noChildrenProp)